### PR TITLE
Update User guide (Delete feature)

### LINKED ISSUE #324: Update UG (Delete Feature)

### PR DESCRIPTION
**Pull request by ljiazh3ng**
resolve #324 
This PR updates the UG:
- [x] add warning / notes for delete feature

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -154,11 +154,18 @@ If you no longer need a note, or accidentally create a wrong one, don't worry. Y
 Format: `delete [-t] TITLE` +
 Example: `delete -t Notably/CS2103`
 
-.Demo for the `delete` command
-image:demo/Delete1.png[]
+[IMPORTANT]
+If your target path is a subtree, the whole subtree including the child would be deleted.
```
Comment (kevinputera) [inline docs/UserGuide.adoc]: Let's not use the term "subtree" here? I don't think people will understand what it means (a bit too technical).
Instead, we can phrase it as: "Please keep in mind that deleting a note that contains other notes will cause the whole group of notes to be deleted. You can refer to the demo below for an illustration."

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -154,11 +154,18 @@ If you no longer need a note, or accidentally create a wrong one, don't worry. Y
...
+
+.Demo for the delete command
+image::demo/Delete1.png[]
+
+.Demo for the delete command (after deletion)
+image::demo/Delete2.png[]
+
+[NOTE]
+You can achieve the same result by using a `AbsolutePath` instead. `delete -t /Workspace/Notably`
```
Comment (kevinputera) [inline docs/UserGuide.adoc]: There's a grammatical error here. It should be "an `AbsolutePath`".

What do you think about this rephrasing: "You can achieve the same result as above by using an `AbsolutePath` instead by executing `delete -t /Workspace/Notably`."?

**Review by kevinputera — APPROVED** [review]
LGTM! Some comments below

File: docs/UserGuide.adoc (hunk 1 above)
Comment (ljiazh3ng) [reply docs/UserGuide.adoc]: good point

**Pull request merged**